integrationtests/mayavi/test_mlab_savefig.py

```diff
@@ -0,0 +1,179 @@
+        self.filename = os.path.join(self.temp_dir, "saved_figure.png")
+
+        # this ensures that the temporary directory is removed
+        self.addCleanup(self.remove_tempdir)
+
+    def remove_tempdir(self):
+        shutil.rmtree(self.temp_dir)
+
+    def setup_engine_and_figure(self, engine):
+        # Set up a Engine/OffScreenEngine/... for the test case
+        self.engine = engine
+
+        if not engine.running:
+            engine.start()
+
+        # figure size is set to be small to force re-rendering
+        engine.new_scene(size=(90, 100))
+        self.figure = engine.current_scene
+
+        # the clean up function will close all figures and stop the engine
+        self.addCleanup(self.cleanup_engine, self.engine)
+
+    def cleanup_engine(self, engine):
+        """ Close all scenes in the engine and stop it """
+        scenes = [scene for scene in engine.scenes]
+        for scene in scenes:
+            engine.close_scene(scene)
+        engine.stop()
+
+    def test_savefig(self):
+        """Test if savefig works with auto size, mag and a normal Engine"""
+        self.setup_engine_and_figure(Engine())
+
+        # Set up the scene
+        create_quiver3d()
+
+        # save the figure (magnification is default "auto")
+        savefig(self.filename, figure=self.figure)
+
+        # check
+        self.check_image()
+
+    def test_savefig_with_size(self):
+        """Test if savefig works with given size and a normal Engine"""
+        self.setup_engine_and_figure(Engine())
+
+        # Set up the scene
+        create_quiver3d()
+
+        # save the figure (magnification is default "auto")
+        savefig(self.filename, size=(131, 217), figure=self.figure)
+
+        # check
+        self.check_image((217, 131))
+
+    @unittest.skipIf(os.environ.get("TRAVIS", False),
+                     ("Offscreen rendering is not tested on Travis "
+                      "due to lack of GLX support"))
+    def test_savefig_with_size_offscreen(self):
+        """Test if savefig works with given size on an OffScreenEngine"""
+        self.setup_engine_and_figure(OffScreenEngine())
+
+        # Set up the scene
+        create_quiver3d()
+
+        # save the figure
+        savefig(self.filename, size=(131, 217), figure=self.figure)
+
+        # check
+        self.check_image((217, 131))
+
+    def test_savefig_with_size_and_magnification(self):
+        """Test if savefig works with given size and magnification"""
+        self.setup_engine_and_figure(Engine())
+
+        # Set up the scene
+        create_quiver3d()
+
+        # save the figure
+        savefig(self.filename, size=(131, 217), magnification=2,
+                figure=self.figure)
+
+        # check if the image size is twice as big
+        self.check_image((434, 262))
+
+    @unittest.skipIf(os.environ.get("TRAVIS", False),
+                     ("Offscreen rendering is not tested on Travis "
+                      "due to lack of GLX support"))
```

Comment:
I have not succeeded at testing offscreen rendering on Travis.
The tests that use offscreen rendering yield the following on Travis.
```
Xlib:  extension "RANDR" missing on display ":99.0".
The program 'python' received an X Window System error.
This probably reflects a bug in the program.
The error was 'GLXBadContextTag'.
  (Details: serial 71 error_code 157 request_code 147 minor_code 5)
  (Note to programmers: normally, X errors are reported asynchronously;
   that is, you will receive the error a while after causing it.
   To debug your program, run it with the --sync command line
   option to change this behavior. You can then get a meaningful
   backtrace from your debugger if you break on the gdk_x_error() function.)
```